wire-ios-sync-engine/Source/SessionManager/SessionManager+APIVersionResolver.swift

```diff
@@ -95,23 +95,25 @@ extension SessionManager: APIVersionResolverDelegate {
 
                 // 1. Tear down the user sessions
                 DispatchQueue.main.sync {
```

Comment:
We are waiting for the `dispatchGroup` down the path anyway with `dispatchGroup.wait(forInterval: 5)`, so can this line be `.async` too?

Reply:
In general the documentation states that one should not run sync code on the main queue at all:
https://developer.apple.com/documentation/dispatch/dispatchqueue

<img width="736" alt="Screenshot 2024-04-17 at 10 08 59" src="https://github.com/wireapp/wire-ios/assets/7196958/66a5ca83-9435-4d03-bf56-f6ca9116df64">

Reply:
Yeah I think we can get rid of the `sync()`